src/libs/elektra/kdb.c

```diff
@@ -1733,27 +2244,47 @@ static void elektraSetRollback (Split * split, Key * parentKey)
  *     with ksNew().
  * @pre The @p parentKey Key must be a valid Key, e.g. constructed with
  *     keyNew(). It must not have read-only name, value or metadata.
+ *     The @p parentKey also must not be in the `meta:/` namespace.
+ *
+ * If you pass `NULL` or a key with read-only metadata as @p parentKey,
+ * kdbSet() will fail immediately without doing anything.
+ * If you pass another invalid @p parentKey, or `NULL` as @p ks or @p handle,
+ * kdbSet() will set an error on @p parentKey and then return immediately.
+ *
+ * **Attention**: If you pass a non-NULL @p parentKey with writable metadata, kdbSet() will **always** remove
```

Comment:
The behavior should be same for all kdb* functions.

Please use doxygen highlighting of sections. I don't think this statement specifically is very important though (only for migration it is worth noting).